server/src/main/java/org/apache/druid/server/coordinator/balancer/ReservoirSegmentSampler.java

```diff
@@ -21,28 +21,86 @@
 
 import org.apache.druid.java.util.emitter.EmittingLogger;
 import org.apache.druid.server.coordinator.ServerHolder;
-import org.apache.druid.server.coordinator.loadqueue.SegmentAction;
+import org.apache.druid.server.coordinator.loading.SegmentAction;
 import org.apache.druid.timeline.DataSegment;
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Function;
 
-final class ReservoirSegmentSampler
+public final class ReservoirSegmentSampler
 {
 
   private static final EmittingLogger log = new EmittingLogger(ReservoirSegmentSampler.class);
 
-  static List<BalancerSegmentHolder> getRandomBalancerSegmentHolders(
+  /**
+   * Picks segments currently loading on the given set of servers that can be
+   * moved to other servers for balancing.
+   *
+   * @param serverHolders        Set of historicals to consider for picking segments
+   * @param maxSegmentsToPick    Maximum number of segments to pick
+   * @param broadcastDatasources Segments belonging to these datasources will not
+   *                             be picked for balancing, since they should be
+   *                             loaded on all servers anyway.
+   * @return Iterator over {@link BalancerSegmentHolder}s, each of which contains
+   * a segment picked for moving and the server currently loading it.
+   */
+  public static List<BalancerSegmentHolder> pickMovableLoadingSegmentsFrom(
```

Comment:
nit: the different in name is soooo minute it took me a long time to realize.  One is `ing` and the other is `ed`.

Is there a reason not to expose the 4-argument method as `public` and have the call sites pass in one of the two lambdas?  It looks like there's only a single call-site for each of them except for tests.

That or maybe try to make the names a bit more different from each other.